### PR TITLE
Implement detection of invalid tar files.

### PR DESCRIPTION
**Pull request by bigkevmcd**
This implements functionality that detects invalid tar files and raises an appropriate exception, rather than just iterating over an empty set of files.

This is a breaking change.

### CODE REVIEW

File: lib/archive/tar/minitar.rb [hunk 1]
```diff
@@ -91,6 +91,7 @@ module Archive::Tar::Minitar
   # The exception raised when a file contains a relative path in secure mode
   # (the default for this version).
   SecureRelativePathError = Class.new(Error)
+  InvalidTarStream = Class.new(Error)
```
Comment (halostatue) [inline lib/archive/tar/minitar.rb]: Please put a documentation comment above this new error.

**Comment by bigkevmcd** [comment]
@halostatue thanks, done!

**Comment by bigkevmcd** [comment]
Is there anything I can do to progress this, perhaps an option to decide whether or not to accept invalid tar files?

**Comment by halostatue** [comment]
Unfortunately, what’s needed is my time to be ready to prepare a release based on this.

**Pull request merged**